src/Client/Grpc/GrpcDurableEntityClient.cs

```diff
@@ -54,8 +55,16 @@ public override async Task SignalEntityAsync(
             RequestId = requestId.ToString(),
             Name = operationName,
             Input = this.dataConverter.Serialize(input),
-            ScheduledTime = scheduledTime?.ToTimestamp(),
-        };
+            ScheduledTime = scheduledTime?.ToTimestamp(),
+            RequestTime = DateTimeOffset.UtcNow.ToTimestamp(),
+        };
+
+        if (Activity.Current is { } activity)
+        {
+            request.ParentTraceContext ??= new P.TraceContext();
+            request.ParentTraceContext.TraceParent = activity.Id;
+            request.ParentTraceContext.TraceState = activity.TraceStateString;
```

Comment:
Should we add a null check for `activity.Id` and `activity.TraceStateString`?

Reply:
It looks like these are never null if `Activity.Current` is non-null as per the earlier [discussion](https://github.com/microsoft/durabletask-dotnet/pull/404#discussion_r2035874028) we had